components/tenant-fetcher/internal/authenticator/middleware.go

```diff
@@ -137,27 +137,26 @@ func (a *Authenticator) getKeyFunc() func(token *jwt.Token) (interface{}, error)
 	return func(token *jwt.Token) (interface{}, error) {
 		unsupportedErr := apperrors.NewInternalError("unexpected signing method: %s", token.Method.Alg())
 
-		if token.Method.Alg() == jwt.SigningMethodRS256.Name {
-			a.mux.Lock()
-			keys := a.cachedJWKs.Keys
-			a.mux.Unlock()
-			for _, key := range keys {
-				if key.Algorithm() == token.Method.Alg() {
-					return key.Materialize()
-				}
-			}
-
-			return nil, apperrors.NewInternalError("unable to find key for algorithm %s", token.Method.Alg())
-		}
-
 		switch token.Method.Alg() {
 		case jwt.SigningMethodRS256.Name:
 			a.mux.Lock()
-			keys := a.cachedJWKs.Keys
+			keys := a.cachedJWKs
 			a.mux.Unlock()
-			for _, key := range keys {
+
+			for it := keys.Iterate(context.Background()); it.Next(context.Background()); {
```

Comment:
The same comments from the director apply here.

And a slightly more general question why this is not reused (imported) from the director?

Reply:
For as long as I remember, there were some differences in the methods' use cases back then. That was why the director's middleware methods were not exported into the pkg.